<center> <img src = https://raw.githubusercontent.com/AndreyRysistov/DatasetsForPandas/main/hh%20label.jpg alt="drawing" style="width:400px;">

# <center> Проект: Анализ вакансий из HeadHunter
   

In [1]:
import pandas as pd
import psycopg2

In [2]:
import warnings

warnings.filterwarnings('ignore')

In [4]:
connection = psycopg2.connect(
    dbname=DBNAME,
    user=USER,
    host=HOST,
    password=PASSWORD,
    port=PORT
)

# Юнит 3. Предварительный анализ данных

1. Напишите запрос, который посчитает количество вакансий в нашей базе (вакансии находятся в таблице vacancies). 

In [5]:
# текст запроса
query_3_1 = f'''
SELECT 
  COUNT(name) 
FROM 
  vacancies
'''

In [6]:
df = pd.read_sql_query(query_3_1, connection)
print("Число вакансий в БД (базе данных): {}".format(df.iloc[0][0]))

Число вакансий в БД (базе данных): 49197


2. Напишите запрос, который посчитает количество работодателей (таблица employers). 

In [7]:
query_3_2 = f'''
SELECT 
  COUNT(name) 
FROM 
  employers

'''

In [8]:
df = pd.read_sql_query(query_3_2, connection)
print("Число работодателей в БД: {}".format(df.iloc[0][0]))

Число работодателей в БД: 23501


3. Посчитате с помощью запроса количество регионов (таблица areas).

In [9]:
query_3_3 = f'''
SELECT 
  COUNT(name) 
FROM 
  areas
'''

In [10]:
df = pd.read_sql_query(query_3_3, connection)
print("Количество регионов: {}".format(df.iloc[0][0]))

Количество регионов: 1362


4. Посчитате с помощью запроса количество сфер деятельности в базе (таблица industries).

In [11]:
query_3_4 = f'''
SELECT 
  COUNT(name) 
FROM 
  industries

'''

In [12]:
df = pd.read_sql_query(query_3_4, connection)
print("Количество сфер деятельности: {}".format(df.iloc[0][0]))

Количество сфер деятельности: 294


***

## <center>Юнит 3: Выводы</center>
#### В представленной базе данных содержится 49197 от 23501 работадателей в 1362 регионах и 294 сферах деятельности.

***

# Юнит 4. Детальный анализ вакансий

1. Напишите запрос, который позволит узнать, сколько (cnt) вакансий в каждом регионе (area).
Отсортируйте по количеству вакансий в порядке убывания.

In [13]:
query_4_1 = f'''
SELECT 
  a.name AS "Регион", 
  COUNT(v.name) AS "Число вакансий" 
FROM 
  vacancies AS v 
  JOIN areas AS a ON v.area_id = a.id 
GROUP BY 
  a.name 
ORDER BY 
  2 DESC
'''

In [14]:
df = pd.read_sql_query(query_4_1, connection)
df #Количества вакансий в различных регионах

Регион  Число вакансий
0             Москва            5333
1    Санкт-Петербург            2851
2              Минск            2112
3        Новосибирск            2006
4             Алматы            1892
..               ...             ...
764       Тарко-Сале               1
765    Новоаннинский               1
766            Бирск               1
767           Сасово               1
768             Анна               1

[769 rows x 2 columns]

2. Напишите запрос, чтобы определить у какого количества вакансий заполнено хотя бы одно из двух полей с зарплатой.

In [15]:
query_4_2 = f'''
SELECT 
  COUNT(*) 
FROM 
  vacancies 
WHERE 
  salary_from IS NOT NULL 
  OR salary_to IS NOT NULL
'''

In [16]:
df = pd.read_sql_query(query_4_2, connection)
print("Количество вакансий с заполненным (хотя бы одним) полем с зп: {}".format(df.iloc[0][0]))

Количество вакансий с заполненным (хотя бы одним) полем с зп: 24073


3. Найдите средние значения для нижней и верхней границы зарплатной вилки. Округлите значения до целого.

In [17]:
query_4_3 = f'''
SELECT 
  ROUND(
    AVG(salary_from)
  ) as "Нижняя граница", 
  ROUND(
    AVG(salary_to)
  ) as "Верхняя граница" 
FROM 
  vacancies
'''

In [18]:
df = pd.read_sql_query(query_4_3, connection)
print("Нижняя граница з.п. вилки: {} \n Верхняя граница з.п. вилки {}".format(df.iloc[0][0], df.iloc[0][1]))

Нижняя граница з.п. вилки: 71065.0 
 Верхняя граница з.п. вилки 110537.0


4. Напишите запрос, который выведет количество вакансий для каждого сочетания типа рабочего графика (schedule) и типа трудоустройства (employment), используемого в вакансиях. Результат отсортируйте по убыванию количества.


In [19]:
query_4_4 = f'''
SELECT 
  COUNT(name) AS "Количество вакансий", 
  schedule AS "Тип рабочего графика", 
  employment AS "Тип трудоустройства"
FROM 
  vacancies 
GROUP BY 
  schedule, 
  employment 
ORDER BY 
  COUNT(name) DESC
'''

In [20]:
df = pd.read_sql_query(query_4_4, connection)
df

Количество вакансий Тип рабочего графика  Тип трудоустройства
0                 35367          Полный день     Полная занятость
1                  7802     Удаленная работа     Полная занятость
2                  1593        Гибкий график     Полная занятость
3                  1312     Удаленная работа  Частичная занятость
4                   940       Сменный график     Полная занятость
5                   569          Полный день           Стажировка
6                   367       Вахтовый метод     Полная занятость
7                   347          Полный день  Частичная занятость
8                   312        Гибкий график  Частичная занятость
9                   141          Полный день     Проектная работа
10                  133     Удаленная работа     Проектная работа
11                  116        Гибкий график           Стажировка
12                  101       Сменный график  Частичная занятость
13                   64     Удаленная работа           Стажировка
14                   18        Гибкий график     Проектная работа
15                   12       Сменный график           Стажировка
16                    2       Вахтовый метод     Проектная работа
17                    1       Сменный график     Проектная работа

5. Напишите запрос, выводящий значения поля Требуемый опыт работы (experience) в порядке возрастания количества вакансий, в которых указан данный вариант опыта. 

In [21]:
query_4_5 = f'''
SELECT 
  COUNT(name) AS "Количество вакансий" , 
  experience AS "Требуемый опыт работы"
FROM 
  vacancies 
GROUP BY 
  experience 
ORDER BY 
  COUNT(name) ASC
'''

In [22]:
df = pd.read_sql_query(query_4_5, connection)
df

Количество вакансий Требуемый опыт работы
0                 1337           Более 6 лет
1                 7197             Нет опыта
2                14511         От 3 до 6 лет
3                26152    От 1 года до 3 лет

***

## <center>Юнит 4: Выводы</center>
#### В исследуемой базе данных, наибольшее количество вакансий находится в г.Москва. Среди рассмотренных заработных плат средняя вилка составялется от 71 до 110 т.р.. Наибольшее количество вакансий ориентированы на соискателей ищущих работу на полный день с полной занятностью. Желаемый опыт работы составялет от 1 года до 3 лет.

***

# Юнит 5. Анализ работодателей

1. Напишите запрос, который позволит узнать, какие работодатели находятся на первом и пятом месте по количеству вакансий.

In [23]:
query_5_1 = f'''
SELECT 
  COUNT(v.name) AS "Число вакансий", 
  e.name AS "Работодатель"
FROM 
  vacancies as v 
  JOIN employers as e ON e.id = v.employer_id 
GROUP BY 
  e.name 
ORDER BY 
  COUNT(v.name) DESC 
LIMIT 
  5
'''

In [24]:
df = pd.read_sql_query(query_5_1, connection)
df

Число вакансий   Работодатель
0            1933         Яндекс
1             491     Ростелеком
2             444       Тинькофф
3             428           СБЕР
4             331  Газпром нефть

2. Напишите запрос, который для каждого региона выведет количество работодателей и вакансий в нём.
Среди регионов, в которых нет вакансий, найдите тот, в котором наибольшее количество работодателей.


In [25]:
query_5_2 = f'''
SELECT 
  a.name AS "Регион", 
  COUNT(DISTINCT e.id) AS "Число работодателей", 
  COUNT(DISTINCT v.id) AS "Число вакансий"
FROM 
  areas AS a 
  LEFT JOIN employers e ON a.id = e.area 
  LEFT JOIN vacancies v ON v.area_id = a.id 
WHERE 
  v.id IS NULL 
GROUP BY 
  a.id 
ORDER BY 
  2 DESC
'''



In [26]:
df = pd.read_sql_query(query_5_2, connection)
df

Регион  Число работодателей  Число вакансий
0                Россия                  410               0
1             Казахстан                  207               0
2    Московская область                   75               0
3    Краснодарский край                   19               0
4    Ростовская область                   18               0
..                  ...                  ...             ...
588             Скидель                    0               0
589          Белоозерск                    0               0
590           Орловский                    0               0
591          Павловская                    0               0
592             Таксимо                    0               0

[593 rows x 3 columns]

3. Для каждого работодателя посчитайте количество регионов, в которых он публикует свои вакансии. Отсортируйте результат по убыванию количества.


In [27]:
query_5_3 = f'''
SELECT 
  employer_id AS "Работодатель", 
  COUNT(DISTINCT area_id) AS "Регион" 
FROM 
  vacancies 
GROUP BY 
  employer_id 
ORDER BY 
  COUNT(DISTINCT area_id) DESC
'''

In [28]:
df = pd.read_sql_query(query_5_3, connection)
df

Работодатель  Регион
0              1740     181
1              2748     152
2           5724811     116
3           5130287      88
4           3682876      71
...             ...     ...
14901        810278       1
14902        810313       1
14903        810551       1
14904        810688       1
14905        811902       1

[14906 rows x 2 columns]

4. Напишите запрос для подсчёта количества работодателей, у которых не указана сфера деятельности. 

In [29]:
query_5_4 = f'''
SELECT 
  COUNT(e.id) 
FROM 
  employers AS e 
  LEFT JOIN employers_industries AS ei ON e.id = ei.employer_id 
WHERE 
  ei.industry_id IS NULL
'''


In [30]:
df = pd.read_sql_query(query_5_4, connection)
print("Количество работодателей без сферы деятельности: {}".format(df.iloc[0][0]))

Количество работодателей без сферы деятельности: 8419


5. Напишите запрос, чтобы узнать название компании, находящейся на третьем месте в алфавитном списке (по названию) компаний, у которых указано четыре сферы деятельности. 

In [31]:
query_5_5 = f'''
SELECT 
  e.name AS "Работодатель", 
  COUNT(ei.industry_id) AS "Количетство сфер деятельности" 
FROM 
  employers AS e 
  LEFT JOIN employers_industries AS ei ON e.id = ei.employer_id 
GROUP BY 
  e.name 
HAVING 
  COUNT(ei.industry_id) = '4' 
ORDER BY 
  e.name
LIMIT 1 OFFSET 2
'''

In [32]:
df = pd.read_sql_query(query_5_5, connection)
df

Работодатель  Количетство сфер деятельности
0         2ГИС                              4

6. С помощью запроса выясните, у какого количества работодателей в качестве сферы деятельности указана Разработка программного обеспечения.


In [33]:
query_5_6 = f'''
SELECT 
  COUNT(ei.employer_id) 
FROM 
  employers_industries AS ei 
  LEFT JOIN industries AS i ON i.id = ei.industry_id 
WHERE 
  i.name = 'Разработка программного обеспечения'
'''

In [34]:
df = pd.read_sql_query(query_5_6, connection)
print("Сфер деятельност 'Разработка программного обеспечения' указана у {} работодателей".format(df.iloc[0][0]))

Сфер деятельност 'Разработка программного обеспечения' указана у 3553 работодателей


7. Для компании «Яндекс» выведите список регионов-миллионников, в которых представлены вакансии компании, вместе с количеством вакансий в этих регионах. Также добавьте строку Total с общим количеством вакансий компании. Результат отсортируйте по возрастанию количества.

Список городов-милионников надо взять [отсюда](https://ru.wikipedia.org/wiki/%D0%93%D0%BE%D1%80%D0%BE%D0%B4%D0%B0-%D0%BC%D0%B8%D0%BB%D0%BB%D0%B8%D0%BE%D0%BD%D0%B5%D1%80%D1%8B_%D0%A0%D0%BE%D1%81%D1%81%D0%B8%D0%B8). 

Если возникнут трудности с этим задание посмотрите материалы модуля  PYTHON-17. Как получать данные из веб-источников и API. 

In [35]:
url = 'https://ru.wikipedia.org/wiki/%D0%93%D0%BE%D1%80%D0%BE%D0%B4%D0%B0-%D0%BC%D0%B8%D0%BB%D0%BB%D0%B8%D0%BE%D0%BD%D0%B5%D1%80%D1%8B_%D0%A0%D0%BE%D1%81%D1%81%D0%B8%D0%B8'
df_city = pd.read_html(url)
df_city = df_city[1]
city_list = df_city['Город'].tolist()
city_str = "', '".join(city_list)
city_str = "'" + city_str + "'"

In [36]:
query_5_7 = f'''
SELECT 
  a.name AS "Город-миллионник", 
  COUNT(v.id) AS "Количество вакансий" 
FROM 
  vacancies AS v 
  LEFT JOIN areas AS a ON v.area_id = a.id 
  LEFT JOIN employers AS e ON v.employer_id = e.id 
WHERE 
  e.name = 'Яндекс' 
  AND a.name IN ({city_str}) 
GROUP BY 
  a.name 
UNION 
SELECT 
  'Total' AS "Город-миллионник", 
  COUNT(v.id) AS "Количество вакансий" 
FROM 
  vacancies AS v 
  LEFT JOIN areas AS a ON v.area_id = a.id 
  LEFT JOIN employers AS e ON v.employer_id = e.id 
WHERE 
  e.name = 'Яндекс' 
  AND a.name IN ({city_str}) 
ORDER BY 
  2  
'''

In [37]:
df = pd.read_sql_query(query_5_7, connection)
df

Город-миллионник  Количество вакансий
0              Омск                   21
1         Челябинск                   22
2        Красноярск                   23
3         Волгоград                   24
4             Пермь                   25
5            Казань                   25
6    Ростов-на-Дону                   25
7               Уфа                   26
8            Самара                   26
9         Краснодар                   30
10          Воронеж                   32
11      Новосибирск                   35
12  Нижний Новгород                   36
13     Екатеринбург                   39
14  Санкт-Петербург                   42
15           Москва                   54
16            Total                  485

***

## <center> Юнит 5: Выводы </center>
#### В представенной базе данных, наибольшее количество вакансий у компании "Яндекс". Наибольшее количество вакансий компании "Яндекс" представлено в Москве. 3553 работадатлей ведут деятельность в сфера "Разработка программного обеспечения". 

***

# Юнит 6. Предметный анализ

1. Сколько вакансий имеет отношение к данным?

Считаем, что вакансия имеет отношение к данным, если в её названии содержатся слова 'data' или 'данн'.

*Подсказка: Обратите внимание, что названия вакансий могут быть написаны в любом регистре.* 


In [38]:
query_6_1 = f'''
SELECT 
  COUNT(*) 
FROM 
  vacancies 
WHERE 
  (
    lower(name) LIKE '%data%'
  ) 
  OR (
    lower(name) LIKE '%данн%'
  )
'''

In [39]:
df = pd.read_sql_query(query_6_1, connection)
print("К данным имеет отношение {} вакансий".format(df.iloc[0][0]))

К данным имеет отношение 1771 вакансий


2. Сколько есть подходящих вакансий для начинающего дата-сайентиста? 
Будем считать вакансиями для дата-сайентистов такие, в названии которых есть хотя бы одно из следующих сочетаний:
* 'data scientist'
* 'data science'
* 'исследователь данных'
* 'ML' (здесь не нужно брать вакансии по HTML)
* 'machine learning'
* 'машинн%обучен%'

** В следующих заданиях мы продолжим работать с вакансиями по этому условию.*

Считаем вакансиями для специалистов уровня Junior следующие:
* в названии есть слово 'junior' *или*
* требуемый опыт — Нет опыта *или*
* тип трудоустройства — Стажировка.
 

In [40]:
query_6_2 = f'''
SELECT 
  COUNT(*) 
FROM 
  vacancies 
WHERE 
  (
    name ILIKE '%data scientist%' 
    OR name ILIKE '%data science%' 
    OR name ILIKE '%исследователь данных%' 
    OR (
      name LIKE '%ML%' 
      AND name NOT ILIKE '%HTML%'
    ) 
    OR name ILIKE '%machine learning%' 
    OR name ILIKE '%машинн%обучен%'
  ) 
  AND (
    name ILIKE '%junior%' 
    OR experience = 'Нет опыта' 
    OR employment ILIKE '%стажировка%'
  )
'''

In [41]:
df = pd.read_sql_query(query_6_2, connection)
print("Для начинающего дата-сайентиста подходит(-ят) {} вакансия".format(df.iloc[0][0]))

Для начинающего дата-сайентиста подходит(-ят) 51 вакансия


3. Сколько есть вакансий для DS, в которых в качестве ключевого навыка указан SQL или postgres?

** Критерии для отнесения вакансии к DS указаны в предыдущем задании.*

In [42]:
query_6_3 = f'''
SELECT 
  COUNT(*) 
FROM 
  vacancies 
WHERE 
  (
    name ILIKE '%data scientist%' 
    OR name ILIKE '%data science%' 
    OR name ILIKE '%исследователь данных%' 
    OR (
      name LIKE '%ML%' 
      AND name NOT LIKE '%HTML%'
    ) 
    OR name ILIKE '%machine learning%' 
    OR name ILIKE '%машинн%обучен%'
  ) 
  AND (
    key_skills ILIKE '%SQL%' 
    OR key_skills ILIKE '%postgres%'
  )
'''

In [43]:
df = pd.read_sql_query(query_6_3, connection)
print("В качестве ключевого навыка 'SQL' и 'postgres' указан в {} вакансии(-ях)".format(df.iloc[0][0]))

В качестве ключевого навыка 'SQL' и 'postgres' указан в 201 вакансии(-ях)


4. Проверьте, насколько популярен Python в требованиях работодателей к DS.Для этого вычислите количество вакансий, в которых в качестве ключевого навыка указан Python.

** Это можно сделать помощью запроса, аналогичного предыдущему.*

In [44]:
query_6_4 = f'''
SELECT 
  COUNT(*) 
FROM 
  vacancies 
WHERE 
  (
    name ILIKE '%data scientist%' 
    OR name ILIKE '%data science%' 
    OR name ILIKE '%исследователь данных%' 
    OR (
      name LIKE '%ML%' 
      AND name NOT LIKE '%HTML%'
    ) 
    OR name ILIKE '%machine learning%' 
    OR name ILIKE '%машинн%обучен%'
  ) 
  AND key_skills ILIKE '%Python%'

'''

In [45]:
df = pd.read_sql_query(query_6_4, connection)
print("В качестве ключевого навыка 'Python' указан {} раз".format(df.iloc[0][0]))

В качестве ключевого навыка 'Python' указан 351 раз


5. Сколько ключевых навыков в среднем указывают в вакансиях для DS?
Ответ округлите до двух знаков после точки-разделителя.

In [46]:
query_6_5 = f'''
SELECT 
  ROUND(
    AVG(
      (
        LENGTH(key_skills) - LENGTH(
          REPLACE(key_skills, '\t', '')
        ) + 1
      )
    ), 
    2
  ) as mean_value 
FROM 
  vacancies 
WHERE 
  (
    name ILIKE '%data scientist%' 
    OR name ILIKE '%data science%' 
    OR name ILIKE '%исследователь данных%' 
    OR (
      name LIKE '%ML%' 
      AND name NOT ILIKE '%HTML%'
    ) 
    OR name ILIKE '%machine learning%' 
    OR name ILIKE '%машинн%обучен%'
  ) 
  AND key_skills IS NOT NULL
'''

In [47]:
df = pd.read_sql_query(query_6_5, connection)
print("В вакансиях на соискание должности связанной с DS в среднем указывают {} навыков(-а)".format(df.iloc[0][0]))

В вакансиях на соискание должности связанной с DS в среднем указывают 6.41 навыков(-а)


6. Напишите запрос, позволяющий вычислить, какую зарплату для DS в **среднем** указывают для каждого типа требуемого опыта (уникальное значение из поля *experience*). 

При решении задачи примите во внимание следующее:
1. Рассматриваем только вакансии, у которых заполнено хотя бы одно из двух полей с зарплатой.
2. Если заполнены оба поля с зарплатой, то считаем зарплату по каждой вакансии как сумму двух полей, делённую на 2. Если заполнено только одно из полей, то его и считаем зарплатой по вакансии.
3. Если в расчётах участвует null, в результате он тоже даст null (посмотрите, что возвращает запрос select 1 + null). Чтобы избежать этой ситуацию, мы воспользуемся функцией [coalesce](https://postgrespro.ru/docs/postgresql/9.5/functions-conditional#functions-coalesce-nvl-ifnull), которая заменит null на значение, которое мы передадим. Например, посмотрите, что возвращает запрос `select 1 + coalesce(null, 0)`

Выясните, на какую зарплату в среднем может рассчитывать дата-сайентист с опытом работы от 3 до 6 лет. Результат округлите до целого числа. 

In [48]:
query_6_6 = f'''
SELECT 
  experience AS "Требуемый опыт", 
  ROUND(
    AVG(
      COALESCE(
        (salary_from + salary_to)/2, 
        salary_from, 
        salary_to
      )
    ), 
    0
  ) AS "Средняя з.п." 
FROM 
  vacancies 
WHERE 
  (
    name ILIKE '%data scientist%' 
    OR name ILIKE '%data science%' 
    OR name ILIKE '%исследователь данных%' 
    OR (
      name LIKE '%ML%' 
      AND name NOT ILIKE '%HTML%'
    ) 
    OR name ILIKE '%machine learning%' 
    OR name ILIKE '%машинн%обучен%'
  ) 
  AND (
    salary_from IS NOT NULL 
    OR salary_to IS NOT NULL
  ) 
GROUP BY 
  experience
'''

In [49]:
df = pd.read_sql_query(query_6_6, connection)
df
# print("Средняя зарплата для дата-сайентиста с опытом работы от 3 до 6 лет составляет {} т.р.".format(df.iloc[2][1]))

Требуемый опыт  Средняя з.п.
0           Нет опыта       74643.0
1  От 1 года до 3 лет      139675.0
2       От 3 до 6 лет      243115.0

***

## <center>Юнит 6: Выводы</center>
#### В исследуемой БД, для получения должности дата-сайентиста необходимо обладать в среднем 6-ю навыками, которые засачтую включают в себя знание Python и SQL. При этом стартовая зп будет около 74 т.р., но с дальнейшим развитием она может составить более 200 т.р.

***

# Общий вывод по проекту

#### В рамках выполненного проекта были апробированы навыки написания запросов на языке SQL. Выполнен количественный и качественный анализа БД по вакансиях опуликованным на сайте HeadHunter. В заключении, можно сделать вывод о том, что рынок вакансий ориентированных на работу с данными (по сведениями представленным в БД) ориентирован на соискателей имеющих опыт работы от года и обладающих большим набором ключевых навыков, но при этом работодатлей достойно оплачивают их труд. В качестве дополнительного исследования разумно оценить з.п. по регионам среди вакансий связанных с данными (код ниже).

In [51]:
query_sup = f'''
SELECT 
  a.name AS "Регион", 
  ROUND(
    AVG(
      COALESCE(
        (v.salary_from + v.salary_to)/2, 
        v.salary_from, 
        v.salary_to
      )
    ), 
    0
  ) AS "Средняя з.п." 
FROM 
  vacancies AS v
LEFT JOIN areas AS a ON v.area_id = a.id
WHERE 
  (
    v.name ILIKE '%data scientist%' 
    OR v.name ILIKE '%data science%' 
    OR v.name ILIKE '%исследователь данных%' 
    OR (
      v.name LIKE '%ML%' 
      AND v.name NOT ILIKE '%HTML%'
    ) 
    OR v.name ILIKE '%machine learning%' 
    OR v.name ILIKE '%машинн%обучен%'
  ) 
  AND (
    v.salary_from IS NOT NULL 
    OR v.salary_to IS NOT NULL
  ) 
GROUP BY 
  a.name
ORDER BY 
  2
'''

In [52]:
df = pd.read_sql_query(query_sup, connection)
df

Регион  Средняя з.п.
0    Ростов-на-Дону       50000.0
1            Казань       50000.0
2           Барнаул       60000.0
3             Пермь       75000.0
4       Владивосток       85000.0
5               Уфа       90000.0
6            Алматы       94587.0
7      Екатеринбург      124000.0
8           Королев      130000.0
9            Рязань      136667.0
10      Новосибирск      166333.0
11  Санкт-Петербург      173335.0
12         Белгород      200000.0
13           Москва      207630.0
14           Сербия      233794.0
15           Турция      233794.0
16       Черногория      233794.0
17          Армения      268863.0
18             Кипр      300000.0

#### Как видно, наибольшая з.п. обещается на Кипре. Среди российких регионов наибольшая з.п. В Москве

In [53]:
connection.close()